### ISSUE TITLE
Generic.CodeAnalysis.UselessOverridingMethod false positive

### ISSUE THREAD

**Issue by morozov** [squizlabs/PHP_CodeSniffer#3035]
**Describe the bug**
If a child class overrides a parent method by only changing the default value of the arguments, the override is considered useless while in fact it's not.

**Code sample**
```php
<?php

class A
{
    public function echo($x = 1): void
    {
        echo $x, PHP_EOL;
    }
}

class B extends A
{
    public function echo($x = 2): void
    {
        parent::echo($x);
    }
}

(new A())->echo();
// 1

(new B())->echo();
// 2
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs --sniffs=Generic.CodeAnalysis.UselessOverridingMethod test.php`
3. See error message displayed
```
FILE: test.php
------------------------------------------------------------------------------------------------
FOUND 0 ERRORS AND 1 WARNING AFFECTING 1 LINE
------------------------------------------------------------------------------------------------
 13 | WARNING | Possible useless method overriding detected
    |         | (Generic.CodeAnalysis.UselessOverridingMethod.Found)
------------------------------------------------------------------------------------------------
```

**Expected behavior**
The warning should not be reported.

**Versions (please complete the following information):**
 - OS: Linux
 - PHP: 7.4
 - PHPCS: 3.5.5

**Comment by gsherwood** [comment]
This sniff only looks at one file at a time. While it would be able to detect this very exact case, it wouldn't be able to do so if you put your class definitions into different files. This is why the sniff only ever produces warnings; it's pointing out where you might want to check during a manual code review.

I'm going to close this as it's not something the sniff will ever be capable of doing. You may prefer to use a different static analysis tool for this sort of checking and leave PHPCS to look more at code formatting.

**Comment by morozov** [comment]
Thanks for the explanation, @gsherwood. I agree with the resolution.

Just for the reference, https://github.com/doctrine/dbal/issues/4281 is another case where just calling the parent method is no a no-op. It opts the caller out of using strict scalar types.
